packages/flutter/lib/src/cupertino/colors.dart

```diff
@@ -543,41 +541,105 @@ class CupertinoColors {
+/// [DefaultTextStyle]), the widget will automatically resolve the color using
+/// [CupertinoDynamicColor.resolve] against its own [BuildContext], on a best-effort
+/// basis.
+///
+/// {@tool sample}
+/// The following snippet builds a [CupertinoButton] that appears white in light
+/// mode, and changes automatically to black in dark mode.
+///
+/// ```dart
+/// Builder(builder: (BuildContext context) {
+///   return CupertinoButton(
+///     child: child,
+///     // CupertinoDynamicColor works out of box in a CupertinoButton.
+///     color: CupertinoDynamicColor.withBrightness(
+///       color: CupertinoColors.white,
+///       darkColor: CupertinoColors.black,
+///     ),
+///     onPressed: () => null,
+///   );
+/// })
+/// ```
+/// {@end-tool}
+///
+/// ### Using a [CupertinoDynamicColor] from a [CupertinoTheme]
+///
+/// When referring to a [CupertinoTheme] color, generally the color will already
+/// have adapted to the ambient [BuildContext], because [CupertinoTheme.of]
+/// implicitly resolves all the colors used in the retrieved [CupertinoThemeData],
+/// before returning it.
+///
+/// {@tool sample}
+/// The following snippet creates a [Container] with the `primaryColor` of the
+/// current theme. If `primaryColor` is a [CupertinoDynamicColor], the container
+/// will be adaptive, thanks to [CupertinoTheme.of]: it will switch to `primaryColor`'s
+/// dark variant once dark mode is turned on, and turns to primaryColor`'s high
+/// contrast variant when [MediaQueryData.highContrast] is requested in the ambient
+/// [MediaQuery], etc.
+///
+/// ```dart
+/// Builder(builder:(BuildContext context) {
+///   // Container is not a Cupertino widget, but CupertinoTheme.of implicitly resolves
+///   // colors used in the retrieved CupertinoThemeData.
+///   return Container(
+///     color: CupertinoTheme.of(context).primaryColor,
+///   );
+/// })
+/// ```
+/// {@end-tool}
+///
+/// ### Manually Resolving a [CupertinoColor]
+///
+/// When used outside of a Cupertino component, or wrapped in an object opaque to
+/// the receiving Cupertino component, you may have to manually resolve the color
+/// using [CupertinoDynamicColor.resolve] before painting. For example, to use a
+/// custom [Border] in a [CupertinoNavigationBar], the colors used in the [Border]
+/// have to be resolved manually before being passed to [CupertinoNavigationBar]'s
+/// constructor.
+///
+/// {@tool sample}
+///
+/// The following snippet demostrates two cases where you have to manually resolve
+/// a [CupertinoDynamicColor].
+///
+/// ```dart
+/// Builder(builder: (BuildContext context) {
+///   final Color resolvedColor = CupertinoDynamicColor.resolve(CupertinoColors.systemBlue, context);
+///   return CupertinoNavigationBar(
+///     // CupertinoNavigationBar does not know how to resolve colors used in
+///     // a Border class.
+///     border: Border(bottom: BorderSide(color: resolvedColor)),
+///   );
+/// })
+/// ```
+///
+/// ```dart
```

Comment:
Does having two ```dart sections in the same {@tool ...} render correctly in dartdocs?

Reply:
it ended up looking like this:
```dart
Builder(builder: (BuildContext context) {
  final Color resolvedColor = CupertinoDynamicColor.resolve(CupertinoColors.systemBlue, context);
  return CupertinoNavigationBar(
    // CupertinoNavigationBar does not know how to resolve colors used in
    // a Border class.
    border: Border(bottom: BorderSide(color: resolvedColor)),
  );
})

// ...

Builder(builder: (BuildContext context) {
  final Color resolvedColor = CupertinoDynamicColor.resolve(CupertinoColors.systemBlue, context);
  // Container is not a Cupertino widget.
  return Container(color: resolvedColor);
})
```
Thanks for pointing this out!